IP6 addresses are not decoded properly

```
=================================== FAILURES ===================================

_____________________________ test_valid[/ip6/::1] _____________________________

addr_str = '/ip6/::1'
    @pytest.mark.parametrize(
        "addr_str",
        ["/ip4/1.2.3.4",
         "/ip4/0.0.0.0",
         "/ip6/::1",
         "/ip6/2601:9:4f81:9700:803e:ca65:66e8:c21",
         "/onion/timaq4ygg2iegci7:1234",
         "/onion/timaq4ygg2iegci7:80/http",
         "/udp/0",
         "/tcp/0",
         "/sctp/0",
         "/udp/1234",
         "/tcp/1234",
         "/sctp/1234",
         "/udp/65535",
         "/tcp/65535",
         "/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSupNKC",
         "/udp/1234/sctp/1234",
         "/udp/1234/udt",
         "/udp/1234/utp",
         "/tcp/1234/http",
         "/tcp/1234/https",
         "/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSupNKC/tcp/1234",
         "/ip4/127.0.0.1/udp/1234",
         "/ip4/127.0.0.1/udp/0",
         "/ip4/127.0.0.1/tcp/1234",
         "/ip4/127.0.0.1/tcp/1234/",
         "/ip4/127.0.0.1/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSupNKC",
         "/ip4/127.0.0.1/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSupNKC/tcp/1234"])  # nopep8
    def test_valid(addr_str):
        ma = Multiaddr(addr_str)
>       assert str(ma) == addr_str.rstrip("/")
E       assert '/ip6/0.0.0.1' == '/ip6/::1'
E         - /ip6/0.0.0.1
E         + /ip6/::1
```

--- a/multiaddr/codec.py
+++ b/multiaddr/codec.py
@@ -150,8 +150,10 @@ def address_string_to_bytes(proto, addr_string):
 
 def address_bytes_to_string(proto, buf):
     from .util import decode_big_endian_16
-    if proto.code in [P_IP4, P_IP6]:
-        return str(IPAddress(int(buf, 16)))
+    if proto.code == P_IP4:
+        return str(IPAddress(int(buf, 16), 4).ipv4())
+    elif proto.code == P_IP6:
+        return str(IPAddress(int(buf, 16), 6).ipv6())
     elif proto.code in [P_TCP, P_UDP, P_DCCP, P_SCTP]:
         return str(decode_big_endian_16(binascii.unhexlify(buf)))
     elif proto.code == P_ONION:

